rs/state_machine_tests/src/lib.rs

```diff
@@ -1,9 +1,23 @@
+use bitcoin::Network;
 use candid::Decode;
 use core::sync::atomic::Ordering;
 use ic_artifact_pool::canister_http_pool::CanisterHttpPoolImpl;
+use ic_btc_adapter::{
+    config::{address_limits, Config as BitcoinAdapterConfig},
+    start_main_event_loop, AdapterState, BlockchainState, GetSuccessorsHandler,
```

Comment:
those are definitely structs I don't want to leak. Let me think how to do it properly

Reply:
It'd be great if we could have this PR by the end of the week so that the Oisy team could use it for their tests.

> The code hasn't been touched for 2-3 years 

Could this be a reason why leaking those structs might not matter that much after all since you aren't really actively working on the code.

Reply:
Fixed.